datafusion/expr/src/expr_schema.rs

```diff
@@ -58,6 +58,60 @@ impl ExprSchemable for Expr {
     ///
     /// Note: [DFSchema] implements [ExprSchema].
     ///
+    /// # Examples
+    ///
+    /// ## Get the type of a single column expression using different schemas
+    ///
+    /// ```
+    /// # use arrow::datatypes::DataType;
+    /// # use datafusion_common::{DFField, DFSchema};
+    /// # use datafusion_expr::{col, ExprSchemable};
+    /// # use std::collections::HashMap;
+    ///
+    /// fn main() {
+    ///   let expr = col("c");
+    ///
+    ///   // Using a schema where the column `c` is of type Utf8
+    ///   let schema = DFSchema::new_with_metadata(
+    ///     vec![DFField::new_unqualified("c", DataType::Utf8, true)],
+    ///     HashMap::new(),
+    ///   )
+    ///   .unwrap();
+    ///   assert_eq!("Utf8", format!("{}", expr.get_type(&schema).unwrap()));
+    ///
+    ///   // Using a schema where the column `c` is of type Int32
+    ///   let schema = DFSchema::new_with_metadata(
+    ///     vec![DFField::new_unqualified("c", DataType::Int32, true)],
+    ///     HashMap::new(),
+    ///   )
+    ///   .unwrap();
+    ///   assert_eq!("Int32", format!("{}", expr.get_type(&schema).unwrap()));
+    /// }
+    /// ```
+    ///
+    /// ## Get the type of an expression that adds 2 columns. Adding an Int32
```

Comment:
I personally suggest we only keep one of the examples in the API docs (as I think the key point of the API is to show using a DF schema as `ExprSchemable`) 

Perhaps we only keep the example of adding two numbers together?